java/vector/src/test/java/org/apache/arrow/vector/file/TestArrowFile.java

```diff
@@ -156,11 +192,36 @@ public void testWriteReadComplex() throws IOException {
         }
       }
     }
+
+    // Read from stream.
+    try (
+        BufferAllocator readerAllocator = allocator.newChildAllocator("reader", 0, Integer.MAX_VALUE);
+        ByteArrayInputStream input = new ByteArrayInputStream(stream.toByteArray());
+        ArrowStreamReader arrowReader = new ArrowStreamReader(input, readerAllocator);
+        BufferAllocator vectorAllocator = allocator.newChildAllocator("final vectors", 0, Integer.MAX_VALUE);
+        MapVector parent = new MapVector("parent", vectorAllocator, null)
+        ) {
+      arrowReader.init();
+      Schema schema = arrowReader.getSchema();
+      LOGGER.debug("reading schema: " + schema);
+
+      try (VectorSchemaRoot root = new VectorSchemaRoot(schema, vectorAllocator)) {
+        VectorLoader vectorLoader = new VectorLoader(root);
+        while (true) {
+          try (ArrowRecordBatch recordBatch = arrowReader.nextRecordBatch()) {
```

Comment:
I wonder if it would be useful to have a common interface for ArrowStreamReader and ArrowReader (or a wrapper for ArrowReader that implements that interface) for iterating through record batches. Would probably make the unit tests share more code.